packages/better-auth/src/oauth2/validate-authorization-code.ts

```diff
@@ -44,10 +44,10 @@ export async function validateAuthorizationCode({
 	// Fixes compatibility with providers like Notion, Twitter, etc.
 	if (authentication === "basic") {
 		const encodedCredentials = base64.encode(
-			`${options.clientId}:${options.clientSecret}`,
+			`${options.clientId}:${options.clientSecret ?? ""}`,
 		);
 		requestHeaders["authorization"] = `Basic ${encodedCredentials}`;
-	} else {
+	} else if (options.clientSecret !== undefined) {
 		body.set("client_secret", options.clientSecret);
 	}
```

Comment:
let's avoid else if...
```suggestion
	}
	if(authentication !== "basic" && options.clientSecret){
	    body.set("client_secret", options.clientSecret);
	}
```